eqcorrscan/core/match_filter/tribe.py

```diff
@@ -353,25 +371,77 @@ def read(self, filename):
             temp_dir = tempfile.mkdtemp()
             arc.extractall(path=temp_dir, members=_safemembers(arc))
             tribe_dir = glob.glob(temp_dir + os.sep + '*')[0]
-            self._read_from_folder(dirname=tribe_dir)
+            self._read_from_folder(
+                dirname=tribe_dir, cores=cores)
         shutil.rmtree(temp_dir)
         return self
 
-    def _read_from_folder(self, dirname):
+    def _read_from_folder(self, dirname, cores=1):
         """
         Internal folder reader.
 
         :type dirname: str
         :param dirname: Folder to read from.
+        :type cores: int
+        :param cores:
+            Maximum number of processes to read waveforms with. Defaults to
+            serial.
         """
+        cores = cores or multiprocessing.cpu_count()
         templates = _par_read(dirname=dirname, compressed=False)
         t_files = glob.glob(dirname + os.sep + '*.ms')
-        tribe_cat_file = glob.glob(os.path.join(dirname, "tribe_cat.*"))
-        if len(tribe_cat_file) != 0:
-            tribe_cat = read_events(tribe_cat_file[0])
+        t_files_dict = {os.path.splitext(os.path.basename(t))[0]: t
+                        for t in t_files}
+        if len(t_files) > len(t_files_dict):
+            Logger.warning("Duplicate template files found")
+        tribe_cat_files = glob.glob(os.path.join(dirname, "tribe_cat.*"))
+        tribe_cat = Catalog()
+        Logger.info("Reading tribe catalog")
+        if cores == 1 or len(tribe_cat_files) == 1:
```

Comment:
I think here it would be useful to include a 
`cores = min(cores, len(tribe_cat_files))`
That would avoid the Pool to start more processes than necessary.

Reply:
See line 404, just below, before the pool is started.

Reply:
Bu you are right, I could just move that higher up and just check `if cores==1`.

Reply:
Oops, sorry should have looked a bit further! I observed that it was using more processes than useful, but now realized that the branch wasn't correctly updated locally.